misc/dbt-materialize/tests/adapter/test_unit_testing.py

```diff
@@ -0,0 +1,32 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License in the LICENSE file at the
+# root of this repository, or online at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dbt.tests.adapter.unit_testing.test_case_insensitivity import (
+    BaseUnitTestCaseInsensivity,
+)
+from dbt.tests.adapter.unit_testing.test_invalid_input import BaseUnitTestInvalidInput
+from dbt.tests.adapter.unit_testing.test_types import BaseUnitTestingTypes
+
+
+class TestMaterializeUnitTestingTypes(BaseUnitTestingTypes):
```

Comment:
Can be done as a follow-up, but we should extend the base `data_types` fixture to include Materialize-specific types that aren't supported in PostgreSQL (e.g. `mz_timestamp`, `map`, `list`).

Reply:
Just extended this to include `mz_timestamp`, however, it seems to be failing with `map` and `list`, most-likely as `array` is not supported by the underlying Postgres adapter yet:
> https://docs.getdbt.com/reference/resource-properties/data-types

Reply:
Added a note to the documentation pointing to the issue that will allow supporting complex types: [`dbt-adapters` #113](https://github.com/dbt-labs/dbt-adapters/issues/113).